modules/models/src/main/java/org/eclipse/xpanse/modules/models/servicetemplate/Deployment.java

```diff
@@ -32,14 +35,21 @@ public class Deployment implements Serializable {
 
     @Valid
     @NotNull
-    @Schema(description = "The variables for the deployment, which will be passed to the deployer")
+    @NotEmpty
+    @UniqueElements
+    @Schema(description = "The variables for the deployment, which will be passed to the deployer."
+            + "The list elements must be unique.")
     private List<DeployVariable> variables;
 
     @Schema(description = "The credential type to do the deployment")
     private CredentialType credentialType = CredentialType.VARIABLES;
 
-    @Schema(description = "The list of availability zones of the service.")
-    private List<AvailabilityZoneConfig> serviceAvailability;
+    @Valid
+    @Size(min = 1)
+    @UniqueElements
+    @Schema(description = "The list of availability zones of the service."
+            + "The list elements must be unique.")
+    private List<AvailabilityZoneConfig> serviceAvailabilities;
```

Comment:
the word service availabilities is not so correct in this context. Even though it is a list, it is more of a configuration that works together to ensure availability of the complete service
We can rename it to serviceAvailbilityConfig or keep it like before.

Reply:
updated.